exchange/bitswap/wantmanager.go

```diff
@@ -17,10 +17,9 @@ import (
 
 type WantManager struct {
 	// sync channels for Run loop
-	incoming   chan *wantSet
-	connect    chan peer.ID        // notification channel for new peers connecting
-	disconnect chan peer.ID        // notification channel for peers disconnecting
-	peerReqs   chan chan []peer.ID // channel to request connected peers on
+	incoming     chan *wantSet
+	connectEvent chan peerStatus     // notification channel for peers connecting/disconnecting
+	peerReqs     chan chan []peer.ID // channel to request connected peers on
```

Comment:
Technically, this bug existed before this PR. However, the whole point of the peerstream update was to fix bugs like this so I figured I might as well fix this now.